### PR TITLE
[CONTP-387] Validation Webhook base e2e test

### PR DESCRIPTION
**Pull request by gabedos**
### What does this PR do?

Tests that the validation admission webhook created by the Datadog Cluster Agent exists in the Kubernetes configuration

### Motivation

E2E test coverage for the validation admission webhook similar to the mutating admission webhook in this previous [PR](https://github.com/DataDog/datadog-agent/pull/30290)

### Describe how to test/QA your changes

Run the `inv new-e2e-tests.run --targets ./tests/containers/ --run TestEKSSuite` command

### Possible Drawbacks / Trade-offs

N/A

### CODE REVIEW

**Review by wdhif — APPROVED** [review]

File: test/new-e2e/tests/containers/k8s_test.go [hunk 1]
```diff
@@ -209,6 +209,20 @@ func (suite *k8sSuite) TestAdmissionControllerWebhooksExist() {
...
+		suite.NotEmpty(validatingConfigs.Items, "No validating webhook configuration found")
+		found := false
+		for _, validatingConfig := range validatingConfigs.Items {
+			if validatingConfig.Name == expectedWebhookName {
+				found = true
+				break
+			}
+		}
+		suite.Require().True(found, fmt.Sprintf("None of the validating webhook configurations have the name '%s'", expectedWebhookName))
+	})
```
Comment (L3n41c) [inline test/new-e2e/tests/containers/k8s_test.go:225]: Why `List`ing all webhooks and iterating over the result to find the expected one?
Could we just try to `Get` the expected one?

```suggestion
	suite.Run("agent registered validating webhook configuration", func() {
		validatingConfig, err := suite.K8sClient.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(ctx, expectedWebhookName, metav1.GetOptions{})
		suite.Require().NoError(err)
		suite.NotNilf(validatingConfig, "None of the validating webhook configurations have the name '%s'", expectedWebhookName)
	})
```

**Review by L3n41c — CHANGES_REQUESTED** [review]
Checking the existence of a webhook by listing all of them and iterating over the result sounds suboptimal.
We can directly try to get the expected one.
The same remark also applies to the `agent registered *mutating* webhook configuration` test committed by #30290.

File: test/new-e2e/tests/containers/k8s_test.go (hunk 1 above)
Comment (gabedos) [reply test/new-e2e/tests/containers/k8s_test.go]: Good catch. I didn't realize that `Get` was in the API. I've also updated the mutating webhook test just above to also use this more concise way. Thanks for calling this out!

**Review by L3n41c — APPROVED** [review]

**Pull request merged**